### PR TITLE
exclude lookup_terms from config errors

### PR DESCRIPTION
**Pull request by bcoca**
##### ISSUE TYPE

  - Bugfix Pull Request
 

##### COMPONENT NAME

config
plugins/lookup
##### ANSIBLE VERSION

```
>=2.6
```

### CODE REVIEW

**Comment by sivel** [comment]
I can confirm at minimum that this resolves https://github.com/ansible/ansible/issues/41696

File: lib/ansible/config/manager.py [hunk 1]
```diff
@@ -365,7 +366,8 @@ def get_config_value_and_origin(self, config, cfile=None, plugin_type=None, plug
                         if plugin_name:
                             entry += 'plugin: %s ' % plugin_name
                     entry += 'setting: %s ' % config
-                    raise AnsibleError("No setting was provided for required configuration %s" % (entry))
+                    if not plugin_type or config not in INTERNAL_DEFS.get(plugin_type):
```
Comment (s-hertel) [inline lib/ansible/config/manager.py]: INTERNAL_DEFS.get(plugin_type, ()) since Nonetype isn't iterable

**Review by s-hertel — COMMENTED** [review]
I'm getting weird behavior for required options in lookups but it doesn't seem like it's due to this. Also doesn't appear that any plugins are broken by it at the moment.
```
diff --git a/lib/ansible/plugins/lookup/vars.py b/lib/ansible/plugins/lookup/vars.py
index 9ce5d30467..8119f110f1 100644
--- a/lib/ansible/plugins/lookup/vars.py
+++ b/lib/ansible/plugins/lookup/vars.py
@@ -18,6 +18,9 @@ DOCUMENTATION = """
         description:
             - What to return if a variable is undefined.
             - If no default is set, it will result in an error if any of the variables is undefined.
+      test_req_options:
+        description: test required options
+        required: true
```
with or without the required option outputs
```
$ ansible -m debug -a msg="{{ lookup('vars', 'foobar', default='ok', test_req_options='test') }}" -c local localhost
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

localhost | FAILED! => {
    "msg": "An unhandled exception occurred while running the lookup plugin 'vars'. Error was a <class 'ansible.errors.AnsibleError'>, original message: Invalid settings supplied for test_req_options: No setting was provided for required configuration plugin_type: lookup plugin: vars setting: test_req_options "
}
```

(Actually, looks like on devel required: True just might be ignored for lookup plugins right now?)

File: lib/ansible/config/manager.py (hunk 1 above)
Comment (bcoca) [reply lib/ansible/config/manager.py]: i don't get that error, weird since it is defined on top in same file .... and plugin_type is not used if None

**Comment by bcoca** [comment]
the other behaviour is because of how i pass 'direct' ... its a bug

**Comment by bcoca** [comment]
try now, that should fix 'direct processing'

File: lib/ansible/config/manager.py (hunk 1 above)
Comment (s-hertel) [reply lib/ansible/config/manager.py]: I got that error using an inventory plugin, not a lookup.

File: lib/ansible/config/manager.py [hunk 2]
```diff
@@ -318,60 +320,68 @@ def get_config_value_and_origin(self, config, cfile=None, plugin_type=None, plug
...
+                # set default if we got here w/o a value
+                if value is None:
+                    if defs[config].get('required', False):
+                        entry = ''
+                        if plugin_type:
+                            entry += 'plugin_type: %s ' % plugin_type
+                            if plugin_name:
+                                entry += 'plugin: %s ' % plugin_name
+                        entry += 'setting: %s ' % config
+                        if not plugin_type or config not in INTERNAL_DEFS.get(plugin_type):
```
Comment (s-hertel) [inline lib/ansible/config/manager.py]: Causes unhelpful error if get() returns None
```
$ cat nmap_inv.yml
plugin: nmap

$ ansible-inventory -i nmap_inv.yml --graph
ERROR! Invalid settings supplied for address: argument of type 'NoneType' is not iterable
```

Comment (bcoca) [reply lib/ansible/config/manager.py]: ah, misunderstood your prev comment

Comment (bcoca) [reply lib/ansible/config/manager.py]: pushing fix, good catch

**Comment by s-hertel** [comment]
Looks good to me, shipit

**Pull request merged**